File: frontend/src/api/companySector.js
/**
 * 기업 섹터 정보 API
 */

const API_BASE_URL = process.env.REACT_APP_API_BASE_URL || 'http://localhost:8000';

/**
 * 기업의 티커로 섹터 정보를 조회
 * @param {string} ticker - 기업 티커 심볼
 * @returns {Promise<Object>} 섹터 정보
 */
export const fetchCompanySector = async (ticker) => {
  try {
    const response = await fetch(`${API_BASE_URL}/companies/${ticker}/sector`);
    
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    
    const data = await response.json();
    
    if (data.status === 'success') {
      return data.data;
    } else {
      throw new Error(data.message || '섹터 정보를 가져오는데 실패했습니다.');
    }
  } catch (error) {
    console.error('Error fetching company sector:', error);
    throw error;
  }
};

File: app/api/company_sector.py
"""
기업 섹터 정보 API
"""
from fastapi import APIRouter, HTTPException
from app.services.company_sector_service import get_company_sector_by_ticker
import logging

router = APIRouter()
logger = logging.getLogger(__name__)

@router.get("/companies/{ticker}/sector")
async def get_company_sector(ticker: str):
    """
    기업의 티커로 섹터 정보를 조회하는 API
    
    Args:
        ticker (str): 기업 티커 심볼
        
    Returns:
        dict: 섹터 정보
    """
    try:
        # 티커를 대문자로 변환
        ticker = ticker.upper()
        
        # 섹터 정보 조회
        result = get_company_sector_by_ticker(ticker)
        
        if result.get("sector") is None and "message" in result:
            raise HTTPException(status_code=404, detail=result["message"])
            
        return {
            "status": "success",
            "data": result
        }
        
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"Error in get_company_sector API for ticker {ticker}: {str(e)}")
        raise HTTPException(status_code=500, detail=f"서버 오류: {str(e)}")
